### ISSUE TITLE
parseAll and readAll produce different results

### ISSUE THREAD

**Issue by gradha** [achesak/nim-csv#1]
I was expecting both procs to provide the same result, but it seems they are different. See the following example:

``` nimrod
import csv

proc test() =
  let
    name = "file.csv"
    input_csv = """
Año;Puesto;Campeonato;Prueba;Categoría;
2013;3;España;Pistola neumática;Veteranos;"""
  let data = parse_all(input_csv, "data", ';')
  echo "string ", data.repr
  echo "file ", read_all(name, name, ';').repr

when isMainModule: test()
```

The `file.csv` contains the same values as `input_csv`, you can grab it form http://dl.dropbox.com/u/145894/t/file.csv. When running this program the output I get is:

```
string 0x1059c4290[0x1059cc230[0x1059ca2d0"A\195\177o", 0x1059ca2f8"Puesto", 0x1059c42c0"Campeonato", 0x1059ca320"Prueba", 0x1059c42f0"Categor\195\173a", 0x1059ca348""], 0x1059cb0e0[0x1059ca370"2013", 0x1059ca398"3", 0x1059c4320"Espa\195\177a", 0x1059cd150"Pistola neum\195\161tica", 0x1059c4350"Veteranos"]]

file 0x1059d3088[0x1059cc500[0x1059ca7f8"A\195\177o", 0x1059ca820"Puesto", 0x1059c4ad0"Campeonato", 0x1059ca848"Prueba", 0x1059c4b00"Categor\195\173a", 0x1059ca870""], 0x1059cc550[0x1059ca898"2013", 0x1059ca8c0"3", 0x1059c4b30"Espa\195\177a", 0x1059cd290"Pistola neum\195\161tica", 0x1059c4b60"Veteranos", 0x1059ca8e8""], nil]
```

The first line is parsed the same, but the file version for the second line provides an empty empty column missing in the string version.

More annoyingly `nil` rows are generated. I think these rows should be discarded when they are at the end.

Sometimes empty rows happen in the input data and these `nil` rows are placed at the end of the results, rather than in their proper sequence. See the following modified example:

``` nimrod
  let
    nil_csv = """

Año;Puesto;Campeonato;Prueba;Categoría;


2013;3;España;Pistola neumática;Veteranos;"""
  let nil_data = parse_all(nil_csv, "data", ';')
  echo "string ", nil_data.repr
  for f in 0 .. <nil_data.len: echo nil_data[f].isNil
```

That will output the following, with the first row being treated as empty data, but other rows appended as nil sequences:

```
string 0x104b2f680[0x104b2ee60[0x104b2ee88""], 0x104b30780[0x104b2eeb0"A\195\177o", 0x104b2eed8"Puesto", 0x104b3b350"Campeonato", 0x104b2ef00"Prueba", 0x104b3b380"Categor\195\173a", 0x104b2ef28""], 0x104b2f6c8[0x104b2ef50"2013", 0x104b2ef78"3", 0x104b3b3b0"Espa\195\177a", 0x104b31390"Pistola neum\195\161tica", 0x104b3b3e0"Veteranos"], nil, nil]

false
false
false
true
true
```

**Comment by achesak** [comment]
The first part, with `parseAll()` and `readAll()` giving different results, happened because Nim's `readFile()` proc seems to append an extra blank line at the end of the file. I made a quick edit, this problem should no longer appear.

My output with the fixed version:

```
string 0x7f6a1e24b290[0x7f6a1e253230[0x7f6a1e2512d0"A\195\177o", 0x7f6a1e2512f8"Puesto", 0x7f6a1e24b2c0"Campeonato", 0x7f6a1e251320"Prueba", 0x7f6a1e24b2f0"Categor\195\173a", 0x7f6a1e251348""], 0x7f6a1e2520e0[0x7f6a1e251370"2013", 0x7f6a1e251398"3", 0x7f6a1e24b320"Espa\195\177a", 0x7f6a1e254150"Pistola neum\195\161tica", 0x7f6a1e24b350"Veteranos"]]

file 0x7f6a1e24bb00[0x7f6a1e253460[0x7f6a1e251758"A\195\177o", 0x7f6a1e251780"Puesto", 0x7f6a1e24bb30"Campeonato", 0x7f6a1e2517a8"Prueba", 0x7f6a1e24bb60"Categor\195\173a", 0x7f6a1e2517d0""], 0x7f6a1e2523f8[0x7f6a1e2517f8"2013", 0x7f6a1e251820"3", 0x7f6a1e24bb90"Espa\195\177a", 0x7f6a1e254290"Pistola neum\195\161tica", 0x7f6a1e24bbc0"Veteranos"]]
```

It also seemed to behave inconsistently when the separator character appears on the end of each row. If you don't want to have a blank item at the end, I added a `skipBlankLast` option to both of the parsing procs. If you set this to `true`, it will remove these blank ones.

The second part is not a problem with this module, as it doesn't deal with blank rows at all. I'm guessing this has something to do with the `parsecsv` module, I'll look at it when I get more time later.

**Comment by achesak** [comment]
The last part of this bug, the nil lines, are fixed as of 9667cdc9066c073c630ccfeb0cc764e895d18e43.
